fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java

```diff
@@ -128,16 +132,17 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
                 loanApplicationTerms.getLoanCalendar(), loanApplicationTerms.getHolidayDetailDTO(), loanApplicationTerms);
 
         if (!scheduleParams.isPartialUpdate()) {
+            Long notDownPaymentPeriodNumber = periods.stream().filter(period -> !period.isDownPaymentPeriod()).count();
             // Set Fixed Principal Amount
-            updateAmortization(mc, loanApplicationTerms, scheduleParams.getPeriodNumber(), scheduleParams.getOutstandingBalance());
+            updateAmortization(mc, loanApplicationTerms, notDownPaymentPeriodNumber.intValue(), scheduleParams.getOutstandingBalance());
```

Comment:
Are you sure this is right?

Reply:
yes, the updateAmortization expects to have only disbursement and repayment periods

Reply:
it gives down payment installments + repayment installments.
Ofc we can refactor the `updateAmortization` itself, but this was the easier and clearer solution